src/main/scala/rise/eqsat/EGraph.scala

```diff
@@ -1,28 +1,39 @@
 package rise.eqsat
 
 object EGraph {
-  def emptyWithAnalysis[D](analysis: Analysis[D]): EGraph[D] = new EGraph(
-    analysis = analysis,
-    memo = HashMap.empty,
-    classes = HashMap.empty[EClassId, EClass[D]],
-    unionFind = UnionFind.empty,
-    pending = Vec.empty,
-    analysisPending = HashSet.empty,
-    classesByMatch = HashMap.empty
-  )
+  def emptyWithAnalysis[ED, ND, TD](analysis: Analysis[ED, ND, TD])
+  : EGraph[ED, ND, TD] =
+    new EGraph(
+      analysis = analysis,
+      memo = HashMap.empty,
+      classes = HashMap.empty,
+      unionFind = UnionFind.empty,
+      pending = Vec.empty,
+      analysisPending = {
+        import Node.{ordering, natIdOrdering, dataTypeIdOrdering, eclassIdOrdering}
+        collection.mutable.TreeSet.empty[(ENode, EClassId)]
+      },
+      classesByMatch = HashMap.empty,
+      hashConses = HashConses.emptyWithAnalysis(analysis),
+    )
 }
 
 /** A data structure to keep track of equalities between expressions.
   * @see [[https://docs.rs/egg/0.6.0/egg/struct.EGraph.html]]
   */
-class EGraph[Data](
-  val analysis: Analysis[Data],
+class EGraph[ED, ND, TD](
```

Comment:
Can you maybe add a comment to indicate what `ED`, `ND`, and `TD` are (I assume something like Expr Data, Nat Data, Type Data???)

Reply:
done